### PR TITLE
Implement discarding of status changes

### PR DESCRIPTION
**Pull request by mateuszc-kainos**

### CODE REVIEW

**Comment by RogowskaMalgorzata** [comment]
Seems good.

**Comment by rlewan** [comment]
Please resolve the conflicts with master and I can merge 👍

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js [hunk 1]
```diff
@@ -74,6 +72,11 @@ define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/rec
                 }
             );
         };
+
+        $scope.changeSlotDiscardChanges = tdprScheduleService.changeSlotDiscardChanges;
+
+
+
```
Comment (drauf) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js]: please remove these 3 lines

File: src/main/resources/assets/js/application/recruiter/services/tdprScheduleService.js [hunk 2]
```diff
@@ -20,9 +25,14 @@ define(['angular', 'application/recruiter/tdprRecruiterModule'], function (angul
...
             }
-            person.changesPending = true;
         };
 
+        this.changeSlotDiscardChanges = function(personData) {
+            personData.slotsList = angular.copy(personData.oldSlotList);
+            personData.oldSlotList = [];
+            personData.changesPending = false;
+        }
+
```
Comment (drauf) [inline src/main/resources/assets/js/application/recruiter/services/tdprScheduleService.js:35]: also this one

File: src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js [hunk 3]
```diff
@@ -352,6 +352,30 @@ define(['angular', 'angularMocks',
...
+                $scope.changeSlotDiscardChanges(person);
+
+
+                expect(startList).toEqual(person.slotsList);
+
+            });
+        });
+
+
+
```
Comment (drauf) [inline src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js]: and these 2

File: src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js [hunk 4]
```diff
@@ -352,6 +352,30 @@ define(['angular', 'angularMocks',
...
+                person.slotsList = [{
+                    "day": weekStart,
+                    "number": 7,
+                    "type": AvailabilityEnum.available.name
+                }];
+                var startList = angular.copy(person.slotsList);
+
+                $scope.changeSlotTypeCycleThrough(person.slotsList[0], 7, weekStart, person);
+
+
```
Comment (drauf) [inline src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js]: and this one

File: src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js [hunk 5]
```diff
@@ -352,6 +352,30 @@ define(['angular', 'angularMocks',
...
+                $scope.changeSlotDiscardChanges(person);
+
+
```
Comment (drauf) [inline src/main/resources/assets/js/test/recruiter/scheduleServiceTest.js]: and this one

File: src/main/resources/assets/js/application/recruiter/services/tdprScheduleService.js [hunk 6]
```diff
@@ -3,6 +3,11 @@ define(['angular', 'application/recruiter/tdprRecruiterModule'], function (angul
 
         this.changeSlotType = function (slot, slotId, day, person, changeTo) {
             var date = dateFilter(day, "yyyy-MM-dd");
+
+            if (person.changesPending === false || angular.isUndefined(person.changesPending)) {
```
Comment (scyganska) [inline src/main/resources/assets/js/application/recruiter/services/tdprScheduleService.js]: Won't 'if (!person.changesPending)' work?

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js [hunk 7]
```diff
@@ -74,6 +72,11 @@ define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/rec
                 }
             );
         };
+
+        $scope.changeSlotDiscardChanges = tdprScheduleService.changeSlotDiscardChanges;
+
```
Comment (scyganska) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js]: Are those 3 empty lines necessary? ;)

**Pull request merged**